client/app/pages/dashboards/hooks/utils.test.js

```diff
@@ -194,3 +195,56 @@ describe("filterWidgets", () => {
     expect(dashboardAllWidgets).toStrictEqual(result);
   });
 });
+
+describe("calculateLayoutsOrder", () => {
+  test("should return layouts ordered by row and col", () => {
+    const layouts = {
+      "10": { col: 0, row: 0, sizeX: 3, sizeY: 8 },
+      "11": { col: 0, row: 8, sizeX: 3, sizeY: 8 },
+      "12": { col: 0, row: 16, sizeX: 3, sizeY: 4 },
+    };
+
+    const result = calculateLayoutsOrder(layouts);
+    expect(result).toStrictEqual(["10", "11", "12"]);
+  });
+
+  test("should handle empty layouts object", () => {
+    const layouts = {};
+
+    const result = calculateLayoutsOrder(layouts);
+    expect(result).toStrictEqual([]);
+  });
+
+  test("should handle layouts with same row but different col", () => {
+    const layouts = {
+      "10": { col: 0, row: 0, sizeX: 3, sizeY: 8 },
+      "11": { col: 2, row: 0, sizeX: 3, sizeY: 8 },
+      "12": { col: 1, row: 0, sizeX: 3, sizeY: 4 },
+    };
+
+    const result = calculateLayoutsOrder(layouts);
+    expect(result).toStrictEqual(["10", "12", "11"]);
+  });
+
+  test("should handle layouts with same col but different row", () => {
+    const layouts = {
+      "10": { col: 0, row: 0, sizeX: 3, sizeY: 8 },
+      "11": { col: 0, row: 1, sizeX: 3, sizeY: 8 },
+      "12": { col: 0, row: 2, sizeX: 3, sizeY: 4 },
+    };
```

Comment:
Same thing about the order

Reply:
same as [previous comment ](https://github.com/metr-systems/redash/pull/9#discussion_r1818600269)